x/bank/types/codec.go

```diff
@@ -22,6 +22,15 @@ func RegisterInterfaces(registry types.InterfaceRegistry) {
 	)
 
 	msgservice.RegisterMsgServiceDesc(registry, &_Msg_serviceDesc)
+
+	// Register legacy interfaces. These interfaces are only used for migration
+	// scripts.
+	// Deprecated.
+	registry.RegisterInterface(
+		"cosmos.bank.v1beta1.SupplyI",
+		(*SupplyI)(nil),
+		&Supply{},
+	)
```

Comment:
As long as we keep the migration scripts, we need to keep those, which is kind of polluting our interfaceRegistry...

Maybe we could clone a separate interfaceRegistry just for migrations? Not sure.

Reply:
I created https://github.com/cosmos/cosmos-sdk/issues/9004 to discuss this.